### ISSUE TITLE
Add SQL comment highlighting

### ISSUE THREAD

**Issue by codeman101** [vim/vim#3983]
Currently when editing a .sql file with vim only C-Style comments get highlighted as a comment /* like this */ But not the one line style. # like this. Please added the same color of highlighting to the one line style.

**Comment by k-takata** [comment]
Have you contacted the maintainer of the syntax file?

**Comment by codeman101** [comment]
I'm talking about the default syntax highlighting that you get when you install vim. Therefore isn't the maintainer the same was the general editor? (vim) Also I'm not referring to stuff like manually installing jellybeans.

**Comment by chdiza** [comment]
> Therefore isn't the maintainer the same was the general editor?

No.

**Comment by codeman101** [comment]
@chdiza Well that's odd. You'd think reporting it here would be the same thing since vim uses it by default. I don't know how to get in touch with the maintainer other than to post here. I only wanted to report this as a favor to the vim project.

**Comment by k-takata** [comment]
Every syntax syntax file is maintained by each maintainer and the Vim project just includes them.
Issues with the each syntax file should be reported to the each maintainer.

For SQL, see the header of the file:
https://github.com/vim/vim/blob/55c77cf2ea9c15e1ec75d1faf702ec3c9e325271/runtime/syntax/sql.vim#L3

If the maintainer doesn't respond, please report here again.

**Comment by codeman101** [comment]
@k-takata His email address doesn't seem to exist anymore because my email failed to send.

**Comment by andymass** [comment]
@codeman101  did you already read `:help sql`, in particular `:help sql-type-default`?

**Comment by codeman101** [comment]
@andymass No I''m sure I could google how to modify the default style file but that's beside the point from why I made this issue. Thanks anyway.

**Comment by chrisbra** [comment]
if you did not use a custom style, I believe my sqloracle runtime syntax file get's loaded. And it allows to use C like commenting as well as `--` comments. I am not sure, that `#` is actually valid for Oracle (at least, I have never seen it).

**Comment by chrisbra** [comment]
oh and btw: the runtime file would be here: https://github.com/chrisbra/vim-sqloracle-syntax

**Comment by codeman101** [comment]
@chrisbra That's interesting. The reason I even know about the python style commenting being allowed in SQL is because of my experience with MySQL (using the workbench) which is Oracle based. I don't know if SQL Sever supports it though so maybe that's why?

**Comment by chrisbra** [comment]
I believe MySQL and Oracle are quite different (and as far as I know have nothing in common, except that Oracle bought Sun a couple of years ago.) However, I just looked it up, Oracle SQL does not allow to use `#` for comments:
https://docs.oracle.com/cd/B19306_01/appdev.102/b14261/comments.htm

If you want to use the MySQL dialect, use the SQLSetType command. This is all explained in the help `:h ft_sql.txt`.

See also the answer from David (which unfortunately did not get forwarded here): https://groups.google.com/d/msg/vim_dev/NHf_mdEEhSw/i_0RQ_giBQAJ

Closing, as this is clearly an configuration issue.
